build.xml

```diff
@@ -0,0 +1,67 @@
+<project name="qabel-block" basedir="." default="build">
+    <property name="dblog" value="/tmp/postgresql.5433.log"/>
+    <property name="dbdir" value="/tmp/postgresqldata.5433"/>
+    <condition property="dbdir.exists"><available file="${dbdir}" type="dir"/></condition>
+    <condition property="dblog.exists"><available file="${dblog}" type="file"/></condition>
+    <condition property="clean-db.required"><or><available file="${dbdir}" type="dir"/><available file="${dblog}" type="file"/></or></condition>
+
+    <target name="build" depends="install">
+        <antcall target="test"/>
+    </target>
+
+    <target name="clean" depends="clean-db,clean-virtualenv"/>
+
+    <target name="clean-db" if="clean-db.required" depends="clean-dbdir,clean-dblog"/>
+    <target name="clean-dbdir" if="${dbdir.exists}">
+        <delete dir="${dbdir}"/>
+    </target>
+    <target name="clean-dblog" if="${dblog.exists}">
+        <delete file="${dblog}"/>
+    </target>
+
+    <property name="virtualenv" value="${basedir}/venv"/>
+    <property name="vbin" value="${virtualenv}/bin"/>
+    <condition property="virtualenv.exists"><available file="${virtualenv}" type="dir"/></condition>
+    <target name="clean-virtualenv" if="virtualenv.exists">
+        <exec executable="rm" dir="${basedir}">
+            <arg value="-r"/>
+            <arg value="${virtualenv}"/>
+        </exec>
+        <property name="virtualenv.exists" value="false"/>
+    </target>
+    <target name="create-virtualenv">
+        <exec executable="virtualenv">
+            <arg value="${virtualenv}"/>
+            <arg value="--python=python3.5"/>
+            <arg value="--no-site-packages"/>
+        </exec>
+    </target>
+    <target name="virtualenv" depends="create-virtualenv">
+        <exec executable="${vbin}/pip">
+            <arg value="install"/>
+            <arg value="--upgrade"/>
+            <arg value="pip"/>
+            <arg value="setuptools"/>
+            <arg value="wheel"/>
+        </exec>
+    </target>
+    <target name="install" depends="virtualenv">
+        <exec executable="${vbin}/pip">
+            <arg value="install"/>
+            <arg value="-U"/>
+            <arg value="-r"/>
+            <arg value="requirements.txt"/>
+            <arg value="--cache-dir"/>
+            <arg value="~/pip-cache"/>
```

Comment:
It somehow didn't cache correctly, so I implied that it doesn't handly the cache-dir correctly by default. I just need a cache thats there in the next build (=outside of the working directory and outside of the venv).
Possibly, that's not required and I messed up something else... feel free to remove that ;)